pkg/infrastructure/gcp/clusterapi/iam.go

```diff
@@ -48,6 +47,14 @@ func GetWorkerRoles() []string {
 	}
 }
 
+// GetSharedVPCRoles returns the pre-defined roles for a shared VPC installation.
+func GetSharedVPCRoles() []string {
+	return []string{
+		"roles/compute.networkAdmin",
+		"roles/compute.securityAdmin",
```

Comment:
Do we also need Compute Load Balancer Services User ?

Reply:
I don't think so, at least @jianli-wei did not appear to need this in his testing.